### PR TITLE
shader/execution/builtin: Add `cos()` test

### PR DESCRIPTION
**Pull request by ben-clayton**
And untitle the scalar constructor function names.
The CTS has a convention of untitled function names, and this matches the WGSL syntax better anyway.

<hr>

**Author checklist for test code/plans:**

- [x] All outstanding work is tracked with "TODO" in a test/file description or `.unimplemented()` on a test.
- [x] New helpers, if any, are documented using `/** doc comments */` and can be found via `helper_index.txt`.
- [x] (Optional, sometimes not possible.) Tests pass (or partially pass without unexpected issues) in an implementation. (Add any extra details above.)

**[Reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) for test code/plans:** (Note: feel free to pull in other reviewers at any time for any reason.)

- [x] The test path is reasonable, the [description](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) "describes the test, succinctly, but in enough detail that a reader can read only the test plans in a file or directory and evaluate the completeness of the test coverage."
- [x] Tests appear to cover this area completely, except for outstanding TODOs. Validation tests use control cases.
    (This is critical for coverage. Assume anything without a TODO will be forgotten about forever.)
- [x] Existing (or new) test helpers are used where they would reduce complexity.
- [x] TypeScript code is readable and understandable (is unobtrusive, has reasonable type-safety/verbosity/dynamicity).

### CODE REVIEW

File: src/webgpu/shader/execution/builtin/builtin.ts [hunk 1]
```diff
@@ -18,13 +19,50 @@ export interface Comparison {
   expected: string; // The string representation of the 'expected' value (possibly with markup)
 }
 
+/** FloatTest is a function that compares whether the two floating point numbers match. */
+export interface FloatTest {
```
Comment (dneto0) [inline src/webgpu/shader/execution/builtin/builtin.ts:23]: nit: Maybe this should be called FloatMatch ?

File: src/webgpu/shader/execution/builtin/builtin.ts [hunk 2]
```diff
@@ -18,13 +19,50 @@ export interface Comparison {
...
+ */
+export function ulpThreshold(ulp: number): FloatTest {
+  return (got, expected) => {
+    if (got === expected) {
+      return true;
+    }
+    if (!Number.isFinite(got) || !Number.isFinite(expected)) {
+      return false;
+    }
+    return diffULP(got, expected) <= ulp;
```
Comment (dneto0) [inline src/webgpu/shader/execution/builtin/builtin.ts:60]: I wonder if diffULP should be updated to return Infinity when either argument is NaN or Infinity.
Then callers could be simplified, like here.

File: src/webgpu/shader/execution/builtin/builtin.ts [hunk 3]
```diff
@@ -18,13 +19,50 @@ export interface Comparison {
...
+  };
 }
 
 // diff compares 'got' to 'expected, returning the Comparison information.
-function diff(got: Value, expected: Value): Comparison {
+function diff(got: Value, expected: Value, cmpFloats: FloatTest): Comparison {
```
Comment (dneto0) [inline src/webgpu/shader/execution/builtin/builtin.ts:65]: Suggest renaming 'diff' to valuesMatch

File: src/webgpu/shader/execution/builtin/cos.spec.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  )
+  .params(u =>
+    u
+      .combine('storageClass', ['uniform', 'storage_r', 'storage_rw'] as const)
+      .combine('vectorize', [undefined, 2, 3, 4] as const)
+  )
+  .fn(async t => {
+    const cases = new Array<Case>(1000);
+    for (let i = 0; i < cases.length; i++) {
+      const angle = -Math.PI + (2.0 * Math.PI * i) / cases.length;
```
Comment (dneto0) [inline src/webgpu/shader/execution/builtin/cos.spec.ts:33]: This is the easy range for cosine.  Yes, it's periodic, but sometimes the hard part is in doing the range reduction accurately.  Values nearest the largest positive and negative finite values would be interesting too.

Suggestions:
- divide by cases.length-1  so the +Math.PI case is tested too


Please reference https://github.com/gpuweb/cts/issues/792 so this sampling can be visited.  With a reference like that, I'm ok landing this for now.

**Review by dneto0 — CHANGES_REQUESTED** [review]
Please add a link to #792.

Also, naming could be improved.

File: src/webgpu/shader/execution/builtin/cos.spec.ts (hunk 4 above)
Comment (dneto0) [reply src/webgpu/shader/execution/builtin/cos.spec.ts:33]: The reference should be listed as TODO so it can be picked up automatically later.

Comment (ben-clayton) [reply src/webgpu/shader/execution/builtin/cos.spec.ts]: Done.

File: src/webgpu/shader/execution/builtin/builtin.ts (hunk 3 above)
Comment (ben-clayton) [reply src/webgpu/shader/execution/builtin/builtin.ts]: Renamed to `compare`. Hope this works for you.

File: src/webgpu/shader/execution/builtin/builtin.ts (hunk 2 above)
Comment (ben-clayton) [reply src/webgpu/shader/execution/builtin/builtin.ts:57]: Done

File: src/webgpu/shader/execution/builtin/builtin.ts (hunk 1 above)
Comment (ben-clayton) [reply src/webgpu/shader/execution/builtin/builtin.ts]: Done

**Review by dneto0 — APPROVED** [review]
Great. Thanks!

**Pull request merged**